.jenkins/jenkins.sh

```diff
@@ -0,0 +1,217 @@
+#!/bin/bash
+# This is the master script used to trigger Jenkins actions.
+# The idea of this script is to keep the amount of code in the "Execute shell" field small
+#
+# Example syntax:
+# .jenkins/jenkins.sh run_regression_tests
+#
+# Other actions such as test/build/deploy can be defined.
+
+### Some environment variables available from Jenkins
+### Note: for a complete list see https://jenkins.ginko.ch/env-vars.html
+# slave              The name of the build host (daint, kesch, ...).
+# BUILD_NUMBER       The current build number, such as "153".
+# BUILD_ID           The current build id, such as "2005-08-22_23-59-59" (YYYY-MM-DD_hh-mm-ss).
+# BUILD_DISPLAY_NAME The display name of the current build, something like "#153" by default.
+# NODE_NAME          Name of the host.
+# NODE_LABELS        Whitespace-separated list of labels that the node is assigned.
+# JENKINS_HOME       The absolute path of the data storage directory assigned on the master node.
+# JENKINS_URL        Full URL of Jenkins, like http://server:port/jenkins/
+# BUILD_URL          Full URL of this build, like http://server:port/jenkins/job/foo/15/
+# JOB_URL            Full URL of this job, like http://server:port/jenkins/job/foo/
+
+exitError()
+{
+    echo "ERROR $1: $3" 1>&2
+    echo "ERROR     LOCATION=$0" 1>&2
+    echo "ERROR     LINE=$2" 1>&2
+    exit $1
+}
+
+# echo basic setup
+echo "####### executing: $0 $* (PID=$$ HOST=$HOSTNAME TIME=`date '+%D %H:%M:%S'`)"
+
+# start timer
+T="$(date +%s)"
+
+# check sanity of environment
+test -n "$1" || exitError 1001 ${LINENO} "must pass an argument"
+test -n "${slave}" || exitError 1005 ${LINENO} "slave is not defined"
+
+# GTC backend name fix: passed as gtc_gt_* but their real name are gtc:gt:*
+#                       OR gtc_* but their real name is gtc:*
+input_backend="$2"
+if [[ $input_backend = gtc_gt_* ]] ; then
+    # sed explained: replace _ with :, two times
+    input_backend=`echo $input_backend | sed 's/_/:/;s/_/:/'`
+fi
+if [[ $input_backend = gtc_* ]] ; then
+    # sed explained: replace _ with :
+    input_backend=`echo $input_backend | sed 's/_/:/'`
+fi
+
+
+# Read arguments
+action="$1"
+backend="$input_backend"
+experiment="$3"
+
+# check presence of env directory
+pushd `dirname $0` > /dev/null
+envloc=`/bin/pwd`
+popd > /dev/null
+shopt -s expand_aliases
+# Download the env
+. ${envloc}/env.sh
+
+# setup module environment and default queue
+test -f ${envloc}/env/machineEnvironment.sh || exitError 1201 ${LINENO} "cannot find machineEnvironment.sh script"
+. ${envloc}/env/machineEnvironment.sh
+export python_env=${python_env}
+echo "PYTHON env ${python_env}"
+# get root directory of where jenkins.sh is sitting
+export jenkins_dir=`dirname $0`
+
+
+if [ -z "${GT4PY_VERSION}" ]; then
+    export GT4PY_VERSION=`cat GT4PY_VERSION.txt`
+fi
+# If the backend is a GTC backend we fetch the caches
+if [[ $backend != *numpy* ]];then
+    . ${jenkins_dir}/actions/fetch_caches.sh $backend $experiment
```

Comment:
We do not currently have a cache plan for physics, but the jenkins test only run numpy backend. Will include this when we set up the caching plan.